gcp_variant_transforms/testing/integration/run_tests.py

```diff
@@ -0,0 +1,254 @@
+  parser.add_argument('--temp_location', required=True)
+  parser.add_argument('--logging_location', required=True)
+  return parser.parse_args()
+
+
+def _get_test_configs():
+  """Gets all test configs in integration directory and subdirectories."""
+  test_configs = []
+  test_file_path = os.path.join(
+      os.getcwd(), 'gcp_variant_transforms/testing/integration')
+  for root, _, files in os.walk(test_file_path):
+    for filename in files:
+      if filename.endswith('.json'):
+        test_configs.append(_load_test_config(os.path.join(root, filename)))
+  return test_configs
+
+
+def _load_test_config(filename):
+  """Loads an integration test JSON object from a file."""
+  with open(filename, 'r') as f:
+    test = json.loads(f.read())
+    _validate_test(test, filename)
+    return test
+
+
+def _validate_test(test, filename):
+  required_keys = ['test_name', 'table_name', 'input_pattern']
+  for key in required_keys:
+    if key not in test:
+      raise ValueError('Test case in {} is missing required key: {}'.format(
+          filename, key))
+
+
+def _create_tests(test_case_configs, context):
+  return [TestCase(context, **config) for config in test_case_configs]
+
+
+def _create_test(test_case_config, context):
+  """Creates a TestCase from test config."""
+  return TestCase(context, **test_case_config)
+
+
+def _run_test(test, context):
+  return test.run(context)
+
+
+def _handle_failure(response):
+  """Raises errors if test case failed."""
+  if 'error' in response:
+    if 'message' in response['error']:
+      raise TestCaseFailure(response['error']['message'])
+    else:
+      # This case should never happen.
+      raise TestCaseFailure(
+          'No traceback. See logs for more information on error.')
+
+
+def _print_errors(results):
+  """Prints results of test cases and tracebacks for any errors."""
+  errors = []
+  for result, test in results:
+    print '{} ...'.format(test.get_name()),
+    try:
+      error = result.get()
+      print 'ok'
+    except TestCaseFailure as e:
+      print 'FAIL'
+      errors.append((test.get_name(), _get_traceback(str(e))))
+  for test_name, error in errors:
+    print _get_failure_message(test_name, error)
+
+
+def _get_traceback(message):
+  traceback_index = message.find('Traceback')
+  if traceback_index == -1:
+    # If error contains no traceback, provide the message for some context.
+    return message
+  return message[traceback_index:]
+
+
+def _get_failure_message(test_name, message):
+  """Prints a formatted message for failed tests."""
+  lines = [
+      '=' * 70,
+      'FAIL: {}'.format(test_name),
+      '-' * 70,
+      message,
+  ]
+  return '\n' + '\n'.join(lines) + '\n'
+
+
+def main():
+  test_case_configs = _get_test_configs()
+  with TestContextManager(_get_args()) as context:
+    pool = multiprocessing.Pool(processes=len(test_case_configs))
+    results = []
+    for config in test_case_configs:
+      test = _create_test(config, context)
```

Comment:
nit: I think it's more readable to just inline _create_test here (it's a one line method now).

Reply:
Done.